cmd/sniproxy/sniproxy_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package main
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestPortForwardingArguments(t *testing.T) {
+	tests := []struct {
+		in      string
+		wanterr string
+		want    *PortForward
+	}{
+		{"", "", nil},
+		{"bad port specifier", "cannot parse", nil},
```

Comment:
Some other cases to consider: `"tcp/pdq/example.com"` (non-numeric port); `"tcp/2112/"` (empty host).

Reply:
Added cases.